latex/architecture.tex

```diff
@@ -1044,12 +1044,52 @@ \section{Memory objects}
 Elements of a buffer object can be a scalar data type (such as an int, float),
 vector data type, or a user-defined structure. In SYCL, a \gls{buffer} object is a
 templated type (\codeinline{cl::sycl::buffer}), parameterized by the element
-type and number of dimensions. An \gls{image} object is stored in one of a limited
-number of formats. The elements of an image object are selected from a list of
-predefined image formats which are provided by an underlying OpenCL
-implementation.  Images are encapsulated in the \codeinline{cl::sycl::image}
-type, which is templated by the number of dimensions in the image. The minimum
-number of elements in a memory object is one.
+type and number of dimensions.
+
+An \gls{image} is represented by the \codeinline{cl::sycl::image} class
+template, parametrized by the dimensionality. The minimum number of elements in
+a memory object is one. Elements of an \gls{image} are stored in memory
+according to one of a limited number of image formats described by a combination of
+\codeinline{cl::sycl::image_channel_order} and
+\codeinline{cl::sycl::image_channel_type}. An \gls{image} can be constructed
+with any one of these image formats, however whether that \gls{image} can be
+passed to a \gls{sycl-kernel-function} is subject to the target \gls{device}
+supporting that image format. The image formats supported by a \gls{device} is
+implementation defined, however if the \gls{device} is reported as supporting
+images it must support the minimum image formats as described below:
+
+\begin{itemize}
+  \item
+    \codeinline{cl::sycl::image_channel_order::rgba}
+    \begin{itemize}
+      \item
+        \codeinline{cl::sycl::image_channel_type::unorm_int8}
+      \item
+        \codeinline{cl::sycl::image_channel_type::unorm_int16}
+      \item
+        \codeinline{cl::sycl::image_channel_type::signed_int8}
+      \item
+        \codeinline{cl::sycl::image_channel_type::signed_int16}
+      \item
+        \codeinline{cl::sycl::image_channel_type::signed_int32}
+      \item
+        \codeinline{cl::sycl::image_channel_type::unsigned_int8}
+      \item
+        \codeinline{cl::sycl::image_channel_type::unsigned_int16}
+      \item
+        \codeinline{cl::sycl::image_channel_type::unsigned_int32}
+      \item
+        \codeinline{cl::sycl::image_channel_type::fp16}
+      \item
+        \codeinline{cl::sycl::image_channel_type::fp32}
+    \end{itemize}
+  \item
```

Comment:
Where does this `bgra` minimum requirement come from?  I don't see it in OpenCL 1.2 - am I missing it?

Reply:
I see that the full profile requires it but not embedded.

Reply:
I think that we should conform to the embedded profile with this change (no `bgra`).  Either that, or we need to factor profile into the required formats list.

Reply:
SYCL working group: We have opted for clarifying the minimum requirements to match the OpenCL embedded profile, so removing `bgra` from the list.

Reply:
@AerialMantis so I guess we are waiting for you to remove the `bgra` item.